### DISCUSSION TITLE
Relative links

### DISCUSSION THREAD

**Discussion by loranger** [spatie/laravel-export#91] [Q&A] [answered]
Hi,

I tried to export a simple website but my mix assets are using absolute links with the current domain.

How can I avoid this ?

For example
```html
<link href="{{ asset('css/app.css') }}" rel="stylesheet">

...

<script src="{{ asset('js/app.js') }}"></script> 
```

becomes
```html
<link href="http://myproject.docker/css/app.css" rel="stylesheet">

...

<script src="http://myproject.docker/js/app.js"></script>
```

**Comment by ades4827** [comment, marked as answer]
My solution for this problem is not excellent but I have chosen to solve it in this way. It would be a change to better identify which console command was executed but I don't know how to do it.

i added this script in the class AppServiceProvider which uses a new env only for export. In my specific case, I also forced the use of https

`
    public function boot()
    {
        if ($this->app->runningInConsole()) {
            \URL::forceRootUrl(env('APP_URL_EXPORTER'));
            \URL::forceScheme('https');
        }
    }
`

**Reply by loranger** [reply]
Maybe not excellent, but very effective, thanks !

**Comment by secondmanveran** [comment]
This will work temporarily but I think a better approach would be config vars for host and scheme that could be pulled into the package when it's built.

**Comment by rizaldywirawan** [comment]
> Hi,
> 
> I tried to export a simple website but my mix assets are using absolute links with the current domain.
> 
> How can I avoid this ?
> 
> For example
> 
> ```
> <link href="{{ asset('css/app.css') }}" rel="stylesheet">
> 
> ...
> 
> <script src="{{ asset('js/app.js') }}"></script> 
> ```
> 
> becomes
> 
> ```
> <link href="http://myproject.docker/css/app.css" rel="stylesheet">
> 
> ...
> 
> <script src="http://myproject.docker/js/app.js"></script>
> ```

This is so useful! Thank you!

**Comment by mmoollllee** [comment]
Sadly `forceRootUrl` doesn't work with relative URLs. I need this package to generate an offlineVersion to be used by opening index.html from anywhere on the file-system. So theres no URL I could replace this with. Any other chance to think of?
